src/Maths/Silk.NET.Maths/Simd/Simd128.Add.cs

```diff
@@ -0,0 +1,306 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if INTRINSICS
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+#if SSE
+using System.Runtime.Intrinsics.X86;
+#endif
+#if AdvSIMD
+using System.Runtime.Intrinsics.Arm;
+#endif
+
+namespace Silk.NET.Maths
+{
+    public static unsafe partial class Simd128
+    {
+        /// <summary>
+        /// Performs hardware-accelerated Add on 128-bit vectors.
+        /// </summary>
+        [MethodImpl(Scalar.MaxOpt)]
+        public static Vector128<T> Add<T>(Vector128<T> left, Vector128<T> right) where T : unmanaged
+        {
+
+            return Byte(left, right);            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> Byte(Vector128<T> left, Vector128<T> right)
+            {
+                if (typeof(T) == typeof(byte))
+                {
+#if AdvSIMD
+                    if (AdvSimd.IsSupported)
+                    {
+                        return AdvSimd.Add(left.AsByte(), right.AsByte()).As<byte, T>();
+                    }
+#endif
+#if SSE
+                    if (Sse2.IsSupported)
+                    {
+                        return Sse2.Add(left.AsByte(), right.AsByte()).As<byte, T>();
+                    }
+#endif
+#if AVX
+                    if (Avx2.IsSupported)
+                    {
+                        return Avx2.Add(left.AsByte(), right.AsByte()).As<byte, T>();
+                    }
+#endif
```

Comment:
This doesn't make sense - if AVX2 is supported, SSE2 is as well, so the lower branch is never hit - but I'm unsure whether we should prefer SSE or AVX here
Edit: I think usually the AVX method for 128bit is just the same as the SSE one, as the Avx2 type literally inherits from SSE42

Reply:
> This doesn't make sense - if AVX2 is supported, SSE2 is as well, so the lower branch is never hit - but I'm unsure whether we should prefer SSE or AVX here

Hmmmm. So I remove avx then?

Reply:
Yes

Reply:
I better remove sse

Reply:
II don't think that's better, SSE is available on NETCore 3.1, AVX isn't, I think

Reply:
Nevermind, but the check would still prevent some machines that can actually run this from running it, so SSE is preferable

Reply:
Oh, no, I will remove avx for 128 bit